Pcap++/src/DpdkDevice.cpp

```diff
@@ -1092,21 +1092,41 @@ bool DpdkDevice::receivePackets(MBufRawPacket** rawPacketsArr, int& rawPacketArr
 		return false;
 	}
 
+	// Save previous length of provided array in case if it will be reused later.
+	int previousLength = rawPacketArrLength;
+
 	struct rte_mbuf* mBufArray[RX_BURST_SIZE];
 	rawPacketArrLength = rte_eth_rx_burst(m_Id, rxQueueId, mBufArray, RX_BURST_SIZE);
 	LOG_DEBUG("Captured %d packets", rawPacketArrLength);
 
 	if (unlikely(rawPacketArrLength <= 0))
 	{
 		rawPacketArrLength = 0;
-		rawPacketsArr = NULL;
+		if (!reuse)
+		{
+			*rawPacketsArr = NULL;
+		}
 		return true;
 	}
 
 	timeval time;
 	gettimeofday(&time, NULL);
 
-	MBufRawPacket* mBufArr = new MBufRawPacket[rawPacketArrLength];
+	MBufRawPacket* mBufArr = NULL;
+	if (reuse) {

```

Comment:
I'd suggest to add a check that verifies `rawPacketsArr` and `*rawPacketsArr` are not NULL